cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -146,6 +146,62 @@ type Member struct {
 	PeerURL string
 }
 
+// GetMemberID returns a member id of the given peer url
+func (c Client) GetMemberID(peerURL string) (uint64, error) {
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   c.Endpoints,
+		DialTimeout: 30 * time.Second,
+		TLS:         c.TLS,
+	})
+	if err != nil {
+		return 0, err
+	}
+	defer cli.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	resp, err := cli.MemberList(ctx)
+	cancel()
+	if err != nil {
+		return 0, err
+	}
+
+	for _, member := range resp.Members {
+		if member.GetPeerURLs()[0] == peerURL {
+			return member.GetID(), nil
+		}
+	}
+	return 0, nil
+}
+
+// RemoveMember notifies an existing etcd cluster that remove a exist member
+func (c Client) RemoveMember(id uint64) ([]Member, error) {
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   c.Endpoints,
+		DialTimeout: 30 * time.Second,
+		TLS:         c.TLS,
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer cli.Close()
+
+	// Remove a exist member from the cluster
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
```

Comment:
why use 20 here and 10 for `GetMemberID`, also can you use a single constant for all timeouts?

Reply:
@yagonobre 20 timeout is no special meaning, i modify this 10 seconds, consistent with `GetMemberID`. I see all functions about etcd not use single constant, so i don't add constant temporarily. If needs, i will add.